### ISSUE TITLE
Excluded files trigger plugin a new incremental check.

### ISSUE THREAD

**Issue by Melzmr** [TypeStrong/fork-ts-checker-webpack-plugin#558]
## Current behavior
Excluded files trigger plugin a new incremental check.

## Expected behavior
Excluded files must not trigger another incremental check.

## Steps to reproduce the issue
I have the following webpack config parts:
1) `ts-loader`
```
 {
  test: /\.tsx?$/,
  exclude: /node_modules/,
  loader: 'ts-loader',
  options: {
    transpileOnly: true,
    configFile: tsConfigPath,
  }
}
```
2) `fork-ts-checker-webpack-plugin`
```
new ForkTsCheckerWebpackPlugin({
  async: inDev,
  typescript: {
    configFile: tsConfigPath,
  },
  logger: { infrastructure: 'console' },
})
```
3) TS config excludes `dist` folder
```
{
  ...
  "exclude": ["node_modules", "dist"]
}
```

✅ In version 5.2.1 I have the following results after I change a single character in a component in a watch mode:
```
Calling reporter service for incremental check.
Changed files: ["/Users/user1/Desktop/test/testWeb/src/client/pages/Home.tsx"]
Deleted files: []
Hash: 6bd4c7ad42193b7547ca
Version: webpack 4.44.2
Time: 1937ms
Built at: 01/07/2021 1:21:18 AM
Asset      Size  Chunks             Chunk Names
js/server.js  21.3 MiB    main  [emitted]  main
Entrypoint main = js/server.js
[./src/client/pages/Home.tsx] 1.68 KiB {main} [built]
    + 1275 hidden modules
No issues found.
Time: 1992 ms
```

❌ With the same configs in version 6.0.8 I have the following results after I change a single character in a component in a watch mode:
```
Calling reporter service for incremental check.
Changed files: ["/Users/user1/Desktop/test/testWeb/src/client/pages/Home.tsx"]
Deleted files: []
Hash: 3ef157e40a413d3102d7
Version: webpack 4.44.2
Time: 1569ms
Built at: 01/07/2021 1:33:19 AM
 Asset      Size  Chunks             Chunk Names
js/server.js  21.3 MiB    main  [emitted]  main
Entrypoint main = js/server.js
[./src/client/pages/Home.tsx] 1.69 KiB {main} [built]
    + 1275 hidden modules
Issues checking in progress...
No issues found.
Time: 1715 ms
Calling reporter service for incremental check.
Changed files: ["/Users/user1/Desktop/test/testWeb/dist/js/main-ffe2403fd22bca30c789.js"]
Deleted files: []
No issues found.
Time: 168 ms
```

😢 For some reasons generated files from `dist` folder start to trigger another checks.

## Issue reproduction repository
Let me know if reproduction example is surely needed.

## Environment
 - **fork-ts-checker-webpack-plugin**: 6.0.8
 - **typescript**: 4.1.3
 - **eslint**: 7.17.0
 - **webpack**: 4.44.2
 - **os**: macOS 10.15.7
 
 P.S. Let me know if this question should be asked somewhere else and not in github issues and sorry 😅

**Comment by piotr-oles** [comment]
Hi! Thanks for reporting this issue. Please create a reproduction repository :)

**Comment by jimisaacs** [comment]
I can confirm the same issue, as well as additional context. This plugin now reports on unreachable code when as before it didn't when option "allowUnreachableCode" is enabled.

Also the plugin option "typescript.configOverwrite" does not actually work to exclude files. Either with "exclude" or "include".

The only way that actually works with version 6 of the plugin to exclude is issue filtering, or downgrading back to version 5 of the plugin.

**Comment by grumd** [comment]
Can confirm, "excludes" in tsconfig is getting ignored now. Code that was excluded and didn't throw any errors in 5.x will throw errors in 6.x with the same configs.

**Comment by piotr-oles** [comment]
As I said, please create a reproduction repository, so I can help. I have limited resources that I can spend working on this package, so I have to delegate the reproduction step to the community, sorry :)

**Comment by grumd** [comment]
@piotr-oles I've already made a reproduction repository here: https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/580

I guess this issue could be closed as a duplicate?

**Comment by piotr-oles** [comment]
@grumd No, your issue is not related to this one :)

**Comment by Melzmr** [comment]
@piotr-oles Hello again!)

I've decided to give webpack 5 another try and everything is good except this particular issue with incremental checks :(

~Since it's the only problem that stops me from using webpack 5~ _Actually, I just downgraded `fork-ts-checker-webpack-plugin` to `5.2.1` and everything seems to work fine._ I've found some time to create a minimal reproduction example.

[Reproduction repository](https://github.com/Melzmr/excluded-558)

I will be very glad if you find some time and look into this issue. Let me know if I need to change something in the repository.

Thanks!

**Comment by piotr-oles** [comment]
@Melzmr Thanks for the reproduction repository, I found the issue and fixed it in #624 :)

**Comment by Melzmr** [comment]
> @Melzmr Thanks for the reproduction repository, I found the issue and fixed it in #624 :)

Great! I am glad that it helped 🎉

I will give it a try in a while. Thanks
